### DISCUSSION TITLE
Tired of moving workspaces whenever you mess with your monitor?

### DISCUSSION THREAD

**Discussion by BirdeeHub** [i3/i3#5654] [Show and tell]
https://github.com/BirdeeHub/i3autoXrandrMemory

A script for i3 that moves your workspaces back to your monitors based on output when you plug them back in. (requires jq for json parsing from i3-msg)

**Comment by orestisfl** [comment]
Just in case you are not aware, you can assign workspace to specific outputs: https://i3wm.org/docs/userguide.html#workspace_screen

Of course, "remembering" where each workspace was has different benefits.

**Reply by BirdeeHub** [reply]
i was not aware. i want them to move to the primary when the monitor is unplugged, and then move them back when i plug it back in, and thats all i wanted
Does this just do that?

**Reply by BirdeeHub** [reply]
ehh, looking it over, it only sorta does it. I dont want it to specify an initial configuration, only remember.

**Reply by BirdeeHub** [reply]
im happy with my solution but next time i spend hours writing a bash script i should read the docs a bit better next time XD

**Reply by BirdeeHub** [reply]
it works great, i learned about udev, and I wrote it so i like it XD but also setting preferred outputs for workspaces wasnt quite what i wanted anyway, although i may have been satisfied with it had i known about it, or used some of those features to do what i did without fighting json

**Reply by BirdeeHub** [reply]
the fun thing about writing it for yourself is it does exactly what you want and nothing more haha

**Reply by BirdeeHub** [reply]
would have probably been easier to just use the script to set and remember the preferred outputs though rather than fighting json.

**Reply by orestisfl** [reply]
Definitely, you can also use an existing library to make your life easier: https://i3wm.org/docs/ipc.html#_see_also_existing_libraries

**Reply by BirdeeHub** [reply]
oh now that is cool. set i3 stuff from directly within other languages. I like it. I like it a lot actually. another thing i wish i knew rather than torturing myself with bash arrays

**Reply by BirdeeHub** [reply]
for context, i use a laptop and unplug my screen all the time, i just wanted whatever was there to go back lmao

**Reply by BirdeeHub** [reply]
Im also new to i3 and semi new to linux and im really liking it but that was driving me NUTS hahaha

**Comment by BirdeeHub** [comment]
Also, I added a script that just cycles your workspace through the monitors that you can keybind for if you want to just move a workspace. Its a separate standalone script
